src/results.cpp

```diff
@@ -45,18 +45,18 @@ Results::Results(SharedRealm r, Query q, SortDescriptor s, SortDescriptor d)
 
 Results::Results(SharedRealm r, Table& table)
 : m_realm(std::move(r))
-, m_table(&table)
 , m_mode(Mode::Table)
 {
+    m_table.reset(&table);
```

Comment:
It's a bit surprising that it's not possible to construct a `TableRef` from a `Table*`, so we instead have to resort to this default-construct / reset dance.